caravel/templates/index.html

```diff
@@ -3,11 +3,31 @@
 {% block head %}
   {{ super() }}
   <script src="/static/js/continuation.js" type="text/javascript"></script>
+  <script src="/static/js/view.js" type=text/javascript></script>
 {% endblock %}
 {% block content %}
-  <div class="row" id="listings">
-    {% for listing in listings %}{% include "listings/inline.html" %}{%
-      endfor %}
+  <div class = "row">
+      <a href="{{ url_for('new_listing') }}" class="btn btn-default hidden-link">New Listing</a>
+    <div class="btn-group" style="float:right;">
+      {% include "thumbnail_button.html" %}
+      {% include "list_button.html" %}
+    </div>
   </div>
+
+  <hr>
+
+  <div class="row" id="listings">
+    {% if view == "Thumbnail" %}
+      {% for listing in listings %}
+        {%- include "listings/thumbnail.html" -%}
+      {% endfor %}
+    {% else %}
+      {% for listing in listings %}
+        {% include "list_header.html" %}
+        <table class="table">
```

Comment:
I think you still need to futz with this — `<table>` should probably be outside the loop, and the header before it.